accounts-db/src/tiered_storage/hot.rs

```diff
@@ -457,25 +493,148 @@ impl HotStorageWriter {
+        account_data: &[u8],
+        executable: bool,
+        account_hash: Option<&AccountHash>,
+    ) -> TieredStorageResult<usize> {
+        let optional_fields = AccountMetaOptionalFields {
+            rent_epoch: (rent_epoch != Epoch::MAX).then_some(rent_epoch),
+            account_hash: account_hash.map(|account_hash| *account_hash),
+        };
+
+        let mut flags = AccountMetaFlags::new_from(&optional_fields);
+        flags.set_executable(executable);
+
+        let padding_len = padding_bytes(account_data.len());
+        let meta = HotAccountMeta::new()
+            .with_lamports(lamports)
+            .with_account_data_size(account_data.len() as u64)
+            .with_account_data_padding(padding_len)
+            .with_flags(&flags);
+
+        let mut stored_size = 0;
+
+        stored_size += self.storage.write_pod(&meta)?;
+        stored_size += self.storage.write_bytes(account_data)?;
+        stored_size += self
+            .storage
+            .write_bytes(&PADDING_BUFFER[0..(padding_len as usize)])?;
+        stored_size += write_optional_fields(&self.storage, &optional_fields)?;
+
+        Ok(stored_size)
+    }
+
+    /// A work-in-progress function that will eventually implements
+    /// AccountsFile::appends_account()
+    pub fn write_accounts<
+        'a,
+        'b,
+        T: ReadableAccount + Sync,
+        U: StorableAccounts<'a, T>,
+        V: Borrow<AccountHash>,
+    >(
+        &self,
+        accounts: &StorableAccountsWithHashesAndWriteVersions<'a, 'b, T, U, V>,
+        skip: usize,
+    ) -> TieredStorageResult<()> {
+        let mut footer = new_hot_footer();
+        let mut index = vec![];
+        let mut cursor = 0;
+
+        // writing accounts blocks
+        let len = accounts.accounts.len();
+        for i in skip..len {
+            let (account, address, account_hash, _write_version) = accounts.get(i);
+            let index_entry = AccountIndexWriterEntry {
+                address,
+                offset: HotAccountOffset::new(cursor)?,
+            };
+            if let Some(account) = account {
+                cursor += self.write_account(
+                    account.lamports(),
+                    account.rent_epoch(),
+                    account.data(),
+                    account.executable(),
+                    Some(account_hash),
+                )?;
+            } else {
+                // None happens when an account has zero lamports.  If such an
+                // account is included in the input to the write_accounts, it
+                // means we want to persist it to reflect the fact that this
+                // account has turned into a zero-lamport account.
+                cursor += self.write_account(
+                    DEFAULT_ACCOUNT_META.lamports,
+                    DEFAULT_ACCOUNT_META.rent_epoch,
+                    &[],
+                    DEFAULT_ACCOUNT_META.executable,
+                    None,
+                )?;
```

Comment:
Btw, to match what append_vec does, it would be 0 instead of `Epoch::MAX` as the default for u64 should be 0 (I didn't find any default for Epoch when calling unwrap_or_default).  Otherwise, I think it will cause the hash mismatch in later steps when comparing that from an AppendVec.

Reply:
Yep, AccountMeta doesn't provide custom default:

```
 #[derive(Serialize, Deserialize, Clone, Debug, Default, Eq, PartialEq)]
 #[repr(C)]
 pub struct AccountMeta {
     /// lamports in the account
     pub lamports: u64,
     /// the epoch at which this account will next owe rent
     pub rent_epoch: Epoch,
     /// the program that owns this account. If executable, the program that loads this account.
     pub owner: Pubkey,
     /// this account's data contains a loaded program (and is now read-only)
     pub executable: bool,
 }
```

Reply:
Good to know this!  Will use Epoch::MAX instead!

Reply:
Done